sycl/source/detail/pi.cpp

```diff
@@ -196,11 +196,18 @@ vector_class<plugin> initialize() {
       std::cerr << "Failed to bind PI APIs to the plugin: " << PluginNames[I]
                 << std::endl;
     }
+    if (useBackend(SYCL_BE_PI_OPENCL) &&
+        PluginNames[I].find("opencl") != std::string::npos) {
+      // Use the OpenCL plugin as the GlobalPlugin
+      GlobalPlugin = std::make_shared<plugin>(PluginInformation);
+    }
+    if (useBackend(SYCL_BE_PI_CUDA) &&
+        PluginNames[I].find("cuda") != std::string::npos) {
+      // Use the CUDA plugin as the GlobalPlugin
+      GlobalPlugin = std::make_shared<plugin>(PluginInformation);
+    }
```

Comment:
I guess you are not using `if-else` here as we might support multiple active plugins in the future?

Should we then at least add an `assert(GlobalPlugin);` to "document" that a global plugin is required?

Reply:
I'm not sure the assert is needed. This function already fails if there is no backend selected or no plugin available.

I also think ideally we shouldn't need GlobalPlugin, and that it's counter to the multiple plugin design, but that's a larger discussion.